### PR TITLE
Reverse a string

### PR DESCRIPTION
**Pull request by alisavch**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module reverse/strings
+
+go 1.16
+
+require (
+	github.com/stretchr/testify v1.7.0 // indirect
+	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
+	golang.org/x/tools v0.1.1 // indirect
+)
```
Comment (papahmsolo) [inline go.mod:9]: try `go mod tidy` this looks like a lot of redundant dependencies

File: actions_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	if actual != expected {
+		t.Errorf("Expected string %v does not reverse to %v", input, expected)
+	}
+}
+
+func TestSayHelloReverse(t *testing.T) {
+	greeting := Reverse("Hello")
+	assert.Equal(t, "olleH", greeting)
+}
```
Comment (papahmsolo) [inline actions_test.go:31]: Not sure why do we have two distinct functions

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Looks good! Let's improve a bit:

- let's rename `actions.go` to something like `strings.go`
- for simplicity reasons can we drop the `testify` and use just std lib for now as you do?
- let's first merge the reverse functionality and release it as 1.0.0 and only then add capitalize

File: README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+1. Implement a function that reverses a string. Cover your code with unit tests, use TDD
+   approach. Create a package for all the source files. Put your package in a module.
+   Publish your solution to a remote repo. Release the module under version 1.0
+2. Implement a function that capitalizes a string. Add examples for both functions to the test
+   file. Release the module under version 1.1.
+
+Concepts
+- Unit testing, table driven tests, TDD, Examples
+- Packages, Modules, SemVer
+- Strings in Go, UTF-8, Unicode
```
[the remark this answers is not in the document]
Comment (alisavch) [reply README.md]: Ok. thanks, changed

File: actions.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package strings
+
+import "strings"
+
+// Reverse string
```
[the remark this answers is not in the document]
Comment (alisavch) [reply actions.go]: Good point, thanks

File: actions_test.go (hunk 2 above)
Comment (alisavch) [reply actions_test.go]: Really, thanks

File: strings.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Package strings implements functions for manipulating with string
+package strings
+
+// Reverse takes an input string and returns result that reverses the input string
+func Reverse(str string) (result string) {
+	for _, value := range str {
+		result = string(value) + result
+	}
+	return
+}
```
Comment (papahmsolo) [inline strings.go:10]: - not really a problem, but named return doesn't provide any additional details, so we might just get rid of it.
- it's idiomatic in Go to give short names to variables, like `k, v` for key and value in `range`, check out [this](https://github.com/golang/go/wiki/CodeReviewComments#variable-names)
- can we not use a naked return and rather return the value explicitly?

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Well done 👍 

Let's address the last minor comments and we can get the first PR merged!

File: strings.go (hunk 5 above)
Comment (alisavch) [reply strings.go]: ok thanks

**Review by papahmsolo — APPROVED** [review]
Looks good 👍

**Pull request merged**